cirq-google/cirq_google/transformers/target_gatesets/sycamore_gateset_test.py

```diff
@@ -0,0 +1,351 @@
+        cirq.Z(qubits[2]),
+        cirq.SWAP(qubits[0], qubits[1]),
+        strategy=cirq.InsertStrategy.NEW,
+    )
+    gateset = cirq_google.SycamoreTargetGateset()
+    compiled_circuit1 = cirq.optimize_for_target_gateset(circuit1, gateset=gateset)
+    compiled_circuit2 = cirq.optimize_for_target_gateset(circuit2, gateset=gateset)
+    cirq.testing.assert_same_circuits(compiled_circuit1, compiled_circuit2)
+    assert (
+        len(list(compiled_circuit1.findall_operations_with_gate_type(cirq_google.SycamoreGate)))
+        == 3
+    )
+    cirq.testing.assert_circuits_with_terminal_measurements_are_equivalent(
+        circuit1, compiled_circuit1, atol=1e-7
+    )
+
+
+def test_convert_to_sycamore_gates_fsim():
+    q0, q1 = cirq.LineQubit.range(2)
+    circuit = cirq.Circuit(cirq.FSimGate(theta=np.pi / 2, phi=np.pi / 6)(q0, q1))
+    compiled_circuit = cirq.optimize_for_target_gateset(
+        circuit, gateset=cirq_google.SycamoreTargetGateset()
+    )
+    cirq.testing.assert_same_circuits(circuit, compiled_circuit)
+
+
+def test_single_qubit_gate():
+    q = cirq.LineQubit(0)
+    mat = cirq.testing.random_unitary(2)
+    gate = cirq.MatrixGate(mat, qid_shape=(2,))
+    circuit = cirq.Circuit(gate(q))
+    compiled_circuit = cirq.optimize_for_target_gateset(
+        circuit, gateset=cirq_google.SycamoreTargetGateset()
+    )
+    ops = list(compiled_circuit.all_operations())
+    assert len(ops) == 1
+    assert isinstance(ops[0].gate, cirq.PhasedXZGate)
+    cirq.testing.assert_circuits_with_terminal_measurements_are_equivalent(
+        circuit, compiled_circuit, atol=1e-8
+    )
+
+
+def test_single_qubit_gate_phased_xz():
+    q = cirq.LineQubit(0)
+    gate = cirq.PhasedXZGate(axis_phase_exponent=0.2, x_exponent=0.3, z_exponent=0.4)
+    circuit = cirq.Circuit(gate(q))
+    compiled_circuit = cirq.optimize_for_target_gateset(
+        circuit, gateset=cirq_google.SycamoreTargetGateset()
+    )
+    ops = list(compiled_circuit.all_operations())
+    assert len(ops) == 1
+    assert ops[0].gate == gate
+
+
+def test_unsupported_gate():
+    class UnknownGate(cirq.testing.TwoQubitGate):
+        pass
+
+    q0, q1 = cirq.LineQubit.range(2)
+    circuit = cirq.Circuit(UnknownGate()(q0, q1))
+    with pytest.raises(ValueError, match='Unable to convert'):
+        cirq.optimize_for_target_gateset(
+            circuit, gateset=cirq_google.SycamoreTargetGateset(), ignore_failures=False
+        )
+
+
+def test_nested_unsupported_gate():
+    class UnknownGate(cirq.testing.TwoQubitGate):
+        pass
+
+    q0 = cirq.LineQubit(0)
+    q1 = cirq.LineQubit(1)
+    subcircuit = cirq.FrozenCircuit(UnknownGate()(q0, q1))
+    circuit = cirq.Circuit(cirq.CircuitOperation(subcircuit))
+    with pytest.raises(ValueError, match='Unable to convert'):
+        cirq.optimize_for_target_gateset(
+            circuit, gateset=cirq_google.SycamoreTargetGateset(), ignore_failures=False
+        )
+
+
+def test_unsupported_gate_ignoring_failures():
```

Comment:
nit: Add comment explaining why it's OK to ignore the failure.

Reply:
Ignoring failures can be useful in a variety of different use cases. For example, when this method occurs is called as part of a larger transformer pipeline and the user has subsequent transformation steps to convert any unsupported gates. 

I think the flag name + args are clear enough to describe what they do and answering the "why" is beyond the scope here.

Reply:
Ah my confusion was around how "ignoring failures" is being tested, but realized later that `ignore_failures` defaults to True in `optimize_for_target_gateset()` :) Explicitly setting the parameter in the call here would be helpful in reading the test case, but again very much a nit.